src/main/scala/junctions/crossing.scala

```diff
@@ -91,3 +90,46 @@ object ClockToSignal {
     c2s.io.signal_out
   }
 }
+
+object LevelCrossing {
+  class SynchronizerBackend[T <: Data](typ: T, sync: Int, _clock: Clock) extends Module(Some(_clock)) {
```

Comment:
As for bundles, what if it's a bundle of bools (which is what we use for the tile interrupts)? I don't think there's a way to restrict this that would make it both safe and not annoying.